crates/router/src/utils.rs

```diff
@@ -155,3 +158,51 @@ pub fn to_currency_base_unit_asf64(
 pub fn get_payment_attempt_id(payment_id: impl std::fmt::Display, attempt_count: i16) -> String {
     format!("{payment_id}_{attempt_count}")
 }
+
+#[derive(Debug)]
+pub struct QrImage;
+
+pub trait CreateQrImageUrlFromQrData {
+    fn create_image_data_source_url(
+        &self,
+        _qr_code_data: String,
+    ) -> Result<String, error_stack::Report<common_utils::errors::QrCodeError>>;
+}
```

Comment:
Why is this trait needed? You are not gonna use this for any other type right? You can just use

```rust
impl QrImage {
	pub fn create_image_data_source_url(&self,_qr_code_data:String){}
}
```